From 719d3a4b78c5cc22c19fceb500e60419b8b85d01 Mon Sep 17 00:00:00 2001
From: Martin Ritter <martin.ritter@lmu.de>
Date: Mon, 31 Jul 2017 10:29:45 +0200
Subject: [PATCH] Disable 'from ROOT import *' in python 3.6 (ROOT-8931)

This feature is now broken and leads to a segmentation violation.

One could try fixing it instead of disabling it but the code which is
used to install the lookup handler in the module
`RootModule::SetRootLazyLookup` depends on internal python
implementation details for the dict class which are not part of the
public API. As a consequence keeping this alive will lead to very
fragile code, require continous effort to adapt to internal changes and
lead to very unmaintainable code with a lot of #ifdef handling in place.

As such a clear error message that this does not work is in my opinion
the better alternative.
---
 bindings/pyroot/ROOT.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/bindings/pyroot/ROOT.py b/bindings/pyroot/ROOT.py
index 0b5f515..a6b9fad 100644
--- a/bindings/pyroot/ROOT.py
+++ b/bindings/pyroot/ROOT.py
@@ -412,6 +412,8 @@ class ModuleFacade( types.ModuleType ):
    def __getattr2( self, name ):             # "running" getattr
     # handle "from ROOT import *" ... can be called multiple times
       if name == '__all__':
+         if sys.hexversion >= 0x3060000:
+            raise ImportError('"from ROOT import *" is not supported in Python 3')
          if '__IPYTHON__' in __builtins__:
             import warnings
             warnings.warn( '"from ROOT import *" is not supported under IPython' )
-- 
2.7.4

